src/commands/cmd_query.c

```diff
@@ -146,8 +146,9 @@ int MGraph_Query(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
     RedisModuleBlockedClient *bc =
         RedisModule_BlockClient(ctx, NULL, NULL, NULL, 0);
 
-    QueryContext *context = _queryContext_New(ctx, bc, ast, argv[1]);
+    QueryContext *context = _queryContext_New(bc, ast, argv[1]);
 
+    RedisModule_RetainString(ctx, argv[1]);
```

Comment:
I was under the impression that we do not need to retain argv, did you found out otherwise?